src/createStore.js

```diff
@@ -196,9 +194,9 @@ export default function createStore(reducer, preloadedState, enhancer) {
       isDispatching = false
     }
 
-    var listeners = currentListeners = nextListeners
-    for (var i = 0; i < listeners.length; i++) {
-      listeners[i]()
+    listeners = currentListeners.slice()
```

Comment:
Actually implications are much more than this!

Lets think about the nested calls to dispatch. When you do
```javascript
listeners = currentListeners.slice()
```
you're saying: *no matter who was already notified, we now have to notify everyone about the new change*, and we do not have to notify everyone about every change...
This copy of the array also allows to check whether a listener being unsubscribed was already notified or not about the last state change.

At the same time, the time needed to shallow-copy an array is far less than the one needed to go through useless notifications, function calls, state checks within listeners, and so on.
It is a bit more pretending from a memory efficiency point of view, though.

To improve things we can avoid the use of ```shift()``` and revert back to ```listeners[i]()``` having the value of ```i``` controlled among the different nested dispatches.
I can make this change this evening...